### PR TITLE
Add table tags (table, td, th, tr, colgroup)

### PR DESCRIPTION
**Pull request by dev-ph1l**
Fixes #36399

Added table tags and attributes (span, colspan, scope, rowspan) in order to be able to use tables in tooltips.

### CODE REVIEW

**Comment by mdo** [comment]
@XhmikosR @GeoSot Wondering if there was a reason for not including any of these?

**Comment by XhmikosR** [comment]
No reason, I'm not sure these are so common. We should check what Angular has first.

File: js/src/util/sanitizer.js [hunk 1]
```diff
@@ -78,6 +79,10 @@ export const DefaultAllowlist = {
   sub: [],
   sup: [],
   strong: [],
+  table: ['span'],
```
Comment (patrickhlauke) [inline js/src/util/sanitizer.js:82]: `span` is incorrect here

File: js/src/util/sanitizer.js [hunk 2]
```diff
@@ -56,6 +56,7 @@ export const DefaultAllowlist = {
   b: [],
   br: [],
   col: [],
```
Comment (patrickhlauke) [inline js/src/util/sanitizer.js:58]: add `span` attribute

File: js/src/util/sanitizer.js [hunk 3]
```diff
@@ -78,6 +79,10 @@ export const DefaultAllowlist = {
   sub: [],
   sup: [],
   strong: [],
+  table: ['span'],
+  td: ['colspan', 'rowspan'],
```
Comment (patrickhlauke) [inline js/src/util/sanitizer.js:83]: add `headers` attribute

File: js/src/util/sanitizer.js [hunk 4]
```diff
@@ -78,6 +79,10 @@ export const DefaultAllowlist = {
...
+  tr: [],
+  th: ['scope'],
```
Comment (patrickhlauke) [inline js/src/util/sanitizer.js:85]: add `abbr`, `colspan`, `headers`, `rowspan`

File: js/src/util/sanitizer.js [hunk 5]
```diff
@@ -78,6 +79,10 @@ export const DefaultAllowlist = {
...
   u: [],
```
Comment (patrickhlauke) [inline js/src/util/sanitizer.js:86]: also add elements `thead`, `tbody`, `tfoot`, and `caption` ?

**Review by patrickhlauke — CHANGES_REQUESTED** [review]

**Comment by patrickhlauke** [comment]
in principle, i'm in favour of this PR

**Comment by XhmikosR** [comment]
@GeoSot are these covered by tests? I.e. shouldn't we need to add the new elements to tests?

Otherwise, if you agree, we could land this.

**Comment by XhmikosR** [comment]
Actually, NVM, I just read @patrickhlauke comments.

@dev-ph1l please make a new PR addressing the comments and make sure that the new changes are added in tests.

**Pull request closed without merging**